### DISCUSSION TITLE
Not opening MRs with "Branch is 3 hours old - skipping PR creation"

### DISCUSSION THREAD

**Discussion by dominik-bln** [renovatebot/renovate#14724] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.6.4

### Please select which platform you are using if self-hosting.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

14.8

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

On one project we are seeing just ~3 MRs opened and automerged during the first few renovate runs of its 24hour schedule and after that all PR creations are skipped with "Branch is 3 hours old - skipping PR creation" (as far as I could tell it's always the same "3 hours", but I didn't read the logs of all runs). There are at least 20 rate limited updates on the project. I believe this problem started a few months ago when the team adopted a schedule for renovate.

To be honest, I can't tell whether this is a misconfiguration or some unexpected interaction between configurations (my suspects are `rebaseWhen`, `schedule` and `stabilityDays` and the fact that we run tests only on the MR and not the branch commits). I'm pretty sure we are running into this code path:

https://github.com/renovatebot/renovate/blob/81ebad1cfae007d7829b7317386505f049fcb827/lib/workers/repository/update/pr/index.ts#L246-L274

With this configuration:

```json
{
  "branchPrefix": "ren/",
  "rebaseWhen": "behind-base-branch",
  "extends": [
    "config:base",
    "default:semanticCommits",
    "default:prConcurrentLimit20",
    "default:prHourlyLimitNone",
    "docker:enableMajor"
  ],
  "dependencyDashboard": true,
  "stabilityDays": 2,
  "prCreation": "not-pending",
  "ignorePresets": ["group:vueMonorepo"],
  "ignorePaths": ["docker-compose.override.example.yml"],
  "ignoreDeps": ["sass", "sass-loader", "cypress"],
  "platformAutomerge": true,
  "major": {
    "automerge": false
  },
  "docker": {
    "pinDigests": false
  },
  "npm": {
    "automerge": false,
    "rangeStrategy": "pin",
    "extends": [
      "default:automergePatch"
    ],
    "packageRules": [
      {
        "automerge": true,
        "labels": ["Automerge"],
        "matchDepTypes": ["devDependencies"]
      },
      {
        "groupName": "tiptap packages",
        "matchPackagePatterns": ["^tiptap"]
      },
      {
        "automerge": true,
        "labels": ["Automerge"],
        "matchUpdateTypes": ["patch", "minor"],
        "matchPackagePatterns": ["^@sentry/"]
      }
    ]
  },
  "python": {
    "enabled": true,
    "automerge": false,
    "rangeStrategy": "pin",
    "packageRules": [
      {
        "automerge": true,
        "labels": ["Automerge"],
        "matchDepTypes": ["dev-dependencies"]
      },
      {
        "automerge": true,
        "labels": ["Automerge"],
        "matchUpdateTypes": ["patch", "minor"],
        "matchPackageNames": ["boto3"]
      },
      {
        "automerge": true,
        "labels": ["Automerge"],
        "matchUpdateTypes": ["digest"],
        "matchPackageNames": ["minio/minio"]
      },
      {
        "automerge": true,
        "labels": ["Automerge"],
        "matchUpdateTypes": ["patch", "minor"],
        "matchPackagePatterns": ["^types-"]
      }
    ]
  },
  "pre-commit": {
    "enabled": true
  },
  "timezone": "Europe/Berlin",
  "updateNotScheduled": true,
  "schedule": [
    "after 8pm on Wednesday",
    "before 8pm on Thursday"
  ]
}
```

Any help in debugging this would be appreciated.

### Relevant debug logs

<details><summary>Logs</summary>
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branchName":"master","latestCommitDate":"2022-03-17T08:08:36+00:00","msg":"latest commit","time":"2022-03-17T11:48:58.304Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","prsRemaining":20,"msg":"Calculated maximum PRs remaining this run","time":"2022-03-17T11:49:59.922Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","msg":"PullRequests limit = 20","time":"2022-03-17T11:49:59.922Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","msg":"Calculating branchConcurrentLimit (20)","time":"2022-03-17T11:49:59.922Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","msg":"20 already existing branches found: ren/minio-minio-digest,ren/postgres-digest,ren/<<REDACTED>>-7070-devops-spa-server-5.x,ren/cryptography-36.x,ren/decorator-5.x,ren/petl-1.x,ren/reportlab-3.x,ren/types-decorator-5.x,ren/types-python-dateutil-2.x,ren/types-pytz-2021.x,ren/types-pyyaml-6.x,ren/types-redis-4.x,ren/types-requests-2.x,ren/elastic-apm-6.x,ren/jsonschema-4.x,ren/pytest-mock-3.x,ren/argon2_cffi-21.x,ren/boto3-1.x,ren/django-capture-on-commit-callbacks-1.x,ren/hermescache-0.x","time":"2022-03-17T11:49:59.922Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","msg":"Branch concurrent limit remaining: 0","time":"2022-03-17T11:49:59.922Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branchesRemaining":0,"msg":"Calculated maximum branches remaining this run","time":"2022-03-17T11:49:59.922Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","msg":"Branches limit = 0","time":"2022-03-17T11:49:59.922Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Setting current branch to master","time":"2022-03-17T11:49:59.923Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","branchName":"master","latestCommitDate":"2022-03-17T08:08:36+00:00","msg":"latest commit","time":"2022-03-17T11:50:00.215Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"getBranchPr(ren/minio-minio-digest)","time":"2022-03-17T11:50:00.313Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"findPr(ren/minio-minio-digest, undefined, open)","time":"2022-03-17T11:50:00.313Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"branchExists=true","time":"2022-03-17T11:50:00.313Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"dependencyDashboardCheck=undefined","time":"2022-03-17T11:50:00.313Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"recreateClosed is false","time":"2022-03-17T11:50:00.314Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"findPr(ren/minio-minio-digest, chore(deps): update minio/minio digest to 7661852, !open)","time":"2022-03-17T11:50:00.314Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"prAlreadyExisted=false","time":"2022-03-17T11:50:00.314Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Checking if PR has been edited","time":"2022-03-17T11:50:00.314Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","branchName":"ren/minio-minio-digest","msg":"Branch has not been modified","time":"2022-03-17T11:50:00.400Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Checking schedule(after 8pm on Wednesday,before 8pm on Thursday, Europe/Berlin)","time":"2022-03-17T11:50:00.401Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","timezone":"Europe/Berlin","msg":"Found timezone","time":"2022-03-17T11:50:00.404Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Adjusting now for timezone","time":"2022-03-17T11:50:00.405Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Checking 2 schedule(s)","time":"2022-03-17T11:50:00.406Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","parsedSchedule":{"schedules":[{"t_a":[72000],"d":[4]}],"exceptions":[],"error":-1},"msg":"Checking schedule \"after 8pm on Wednesday\"","time":"2022-03-17T11:50:00.407Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","parsedSchedule":{"schedules":[{"t_b":[72000],"d":[5]}],"exceptions":[],"error":-1},"msg":"Checking schedule \"before 8pm on Thursday\"","time":"2022-03-17T11:50:00.409Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Matches schedule before 8pm on Thursday","time":"2022-03-17T11:50:00.409Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Branch already exists","time":"2022-03-17T11:50:00.409Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"getBranchPr(ren/minio-minio-digest)","time":"2022-03-17T11:50:00.409Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"findPr(ren/minio-minio-digest, undefined, open)","time":"2022-03-17T11:50:00.409Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","isStale":false,"currentBranch":"master","currentBranchSha":"42143a0a77d6f56f28abc565122932d59344bf19","msg":"isBranchStale=false","time":"2022-03-17T11:50:00.570Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Branch is up-to-date","time":"2022-03-17T11:50:00.571Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"isBranchConflicted(master, ren/minio-minio-digest)","time":"2022-03-17T11:50:00.571Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","gitAuthorName":"Renovate Bot","msg":"Setting git author name","time":"2022-03-17T11:50:00.571Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","gitAuthorEmail":"renovate-bot@dominik.berlin","msg":"Setting git author email","time":"2022-03-17T11:50:00.707Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Branch does not need rebasing","time":"2022-03-17T11:50:01.410Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Using reuseExistingBranch: true","time":"2022-03-17T11:50:01.410Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"manager.getUpdatedPackageFiles() reuseExistinbranch=true","time":"2022-03-17T11:50:01.410Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"docker-compose.extractPackageFile()","time":"2022-03-17T11:50:01.508Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"docker-compose.extractPackageFile()","time":"2022-03-17T11:50:01.512Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","packageFile":"docker-compose.yml","depName":"minio/minio","msg":"Branch dep is already updated","time":"2022-03-17T11:50:01.513Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","packageFile":"docker-compose.yml","depName":"minio/minio","msg":"No content changed","time":"2022-03-17T11:50:01.513Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"No package files need updating","time":"2022-03-17T11:50:01.513Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Getting updated lock files","time":"2022-03-17T11:50:01.513Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","packageFiles":["docs/package.json","e2e-test/package.json","frontend/package.json"],"msg":"Writing package.json files","time":"2022-03-17T11:50:01.513Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Writing docs/package-lock.json","time":"2022-03-17T11:50:01.513Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Writing e2e-test/package-lock.json","time":"2022-03-17T11:50:01.633Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Writing frontend/package-lock.json","time":"2022-03-17T11:50:01.728Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Writing any updated package files","time":"2022-03-17T11:50:01.856Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"No updated lock files in branch","time":"2022-03-17T11:50:01.858Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"No files to commit","time":"2022-03-17T11:50:01.858Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Checking if we can automerge branch","time":"2022-03-17T11:50:01.858Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"mergeStatus=no automerge","time":"2022-03-17T11:50:01.858Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Ensuring PR","time":"2022-03-17T11:50:01.858Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"There are 0 errors and 0 warnings","time":"2022-03-17T11:50:01.858Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"getBranchPr(ren/minio-minio-digest)","time":"2022-03-17T11:50:01.859Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"findPr(ren/minio-minio-digest, undefined, open)","time":"2022-03-17T11:50:01.859Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Checking branch combined status","time":"2022-03-17T11:50:01.859Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"resolveBranchStatus(branchName=ren/minio-minio-digest, ignoreTests=false)","time":"2022-03-17T11:50:01.859Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"getBranchStatus(ren/minio-minio-digest)","time":"2022-03-17T11:50:01.859Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Got res with 0 results","time":"2022-03-17T11:50:01.931Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Branch status yellow","time":"2022-03-17T11:50:01.931Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Branch status is: yellow","time":"2022-03-17T11:50:01.931Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Branch status is yellow - checking timeout","time":"2022-03-17T11:50:01.931Z","v":0}
{"name":"renovate","hostname":"runner-kwy8rzi6-project-705-concurrent-0kwfjl","pid":17,"level":20,"logContext":"cTNo-rpd7qWpzfXNpR0rP","repository":"alpha/alpha","branch":"ren/minio-minio-digest","msg":"Branch is 3 hours old - skipping PR creation","time":"2022-03-17T11:50:02.021Z","v":0}
</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment, marked as answer]
`"prCreation": "not-pending"` means "do not open a PR if any checks are still pending". `"stabilityDays": 2` will mean that a branch containing any dependency less than 2 days old will have a pending status check added it by Renovate itself. I'm guessing these two factors are interacting together. If it always says 3 hours for every branch, then that's strange though.

Something else you could do is set [internalChecksFilter=strict](https://docs.renovatebot.com/configuration-options/#internalchecksfilter) so that only versions which have passed `stabilityDays` will be proposed. That way you should never receive a `renovate/stability-days` pending check.

**Reply by dominik-bln** [reply]
I experimented a bit further with the project and branches via the dashboard. 

 We have twenty existing branches without any stability checks on the project:

<img width="693" alt="Screenshot 2022-03-18 at 14 00 43" src="https://user-images.githubusercontent.com/855119/159007643-9cba6d95-a76f-422e-aa63-8676ce4b5176.png">

The first one `django-capture-on-commit-callbacks` in `v1.10.0` was released on January 10th. My expectation would have been for this to have gotten an MR at least yesterday during the schedule and basically all the others as well.

I also tried to force the creation of some new MRs via the dashboard now. On the first run all branches, but no MRs were created, even though the `pybreaker` dependency was last released on March 3rd:

<img width="695" alt="Screenshot 2022-03-18 at 14 01 38" src="https://user-images.githubusercontent.com/855119/159007676-9c5513cc-cda3-4443-b509-4340fc44a4ac.png">

For `pytest` with a status check, I was able to force create the MR. The ones without any status checks are just flip-flopping between "Pending Status" check and "Awaiting Schedule" on the dashboard when I try to force those.

Is it possible that somehow the empty branch pipelines are considered to block the MR creation? We are using a Gitlab feature to only run our tests when an MR is already opened.

**Reply by dominik-bln** [reply]
Or should every branch get the stability check? It looks to me like mainly the Python dependencies don't have the check at all 🤔

**Reply by rarkins** [reply]
> Is it possible that somehow the empty branch pipelines are considered to block the MR creation? We are using a Gitlab feature to only run our tests when an MR is already opened.

Yes, this is quite possible. You should see in the debug logs whether Renovate detects there to be an empty list or not. Such an approach does not seem compatible with `prCreation=not-pending` because it sounds like you've intentionally disabled all tests, so it will always be pending.

**Reply by dominik-bln** [reply]
I've disabled `stabilityDays` and `prCreation=not-pending` for the project now and looks like that is working (will only see fully on Thursday during the schedule).

Just an FYI, the "Merge request Pipelines" feature is pretty crucial on Gitlab CI for some features like running API tests only when files in the `/api` folder have changed. So having some way to define that the status checks should only be run on the already opened MRs or allow MR opening if there are no tests would be nice in that regard. Not sure if that warrants a feature request though, it probably would be pretty inconsistent behaviour with other platforms and we could still define a dummy job on the branch if we really needed the `stabilityDays` feature.

I'm also not 100% certain whether the status checks are working as intended, since JS dependencies always seemed to work. Shouldn't the Python deps get a "successful" stability check like the other branches and not no check at all?

Thank you for your time on this matter.

**Reply by rarkins** [reply]
I think if we know a timestamp for a release then `stabilityDays` should always work (or it's a bug). If we don't have any timestamp then maybe we should treat is as "passed" too - which could be a feature request if so.

**Comment by rarkins** [comment]
By the way, this was not correct to raise as a bug report and should have been raised as a discussion, so I have converted it now. If you're having trouble knowing which to raise, please just raise only discussions from now on and if something turns out to be a bug, then a bug issue can be raised after that.

**Reply by dominik-bln** [reply]
Sorry, I'm mainly a Gitlab user, so wasn't aware of the discussion feature yet. I'll use that in the future.
